MyBlog.Services/Areas/Articles/ArticleService.cs

```diff
@@ -0,0 +1,147 @@
+            .Select(article => _mapper.Map<ArticleDto>(article))
+            .ToListAsync();
+        
+        return articles;
+    }
+
+    public async Task<List<ArticleDto>> GetByTopicAsync(Topic topic)
+    {
+        var articles = await _context.Articles
+            .Where(article => article.Topic == topic)
+            .Select(article => _mapper.Map<ArticleDto>(article))
+            .ToListAsync();
+
+        return articles;
+    }
+
+    public async Task<ArticleDto> GetByIdAsync(int id)
+    {
+        var article = await GetArticleByIdAsync(id);
+
+        var articleDto = _mapper.Map<ArticleDto>(article);
+
+        return articleDto;
+    }
+    
+    public async Task<ArticleDto> CreateAsync(JwtInfoDto userToken, ArticleDtoInput articleInput)
+    {
+        await ThrowIfTitleExistAsync(articleInput);
+        
+        var article = _mapper.Map<Article>(articleInput);
+
+        article.UserId = userToken.Id;
+        
+        await _context.Articles.AddAsync(article);
+        await _context.SaveChangesAsync();
+
+        var articleDto = _mapper.Map<ArticleDto>(article);
+        
+        return articleDto;
+    }
+    
+    public async Task<ArticleDto> UpdateByIdAsync(int id, JwtInfoDto userToken, ArticleDtoInput articleInput)
+    {
+        var article = await GetArticleByIdAsync(id);
+
+        ThrowIfUserHasNotEditAccess(article.UserId, userToken);
+        await ThrowIfTitleExistAsync(articleInput, id);
+        
+        _mapper.Map(articleInput, article);
+
+        article.DateUpdated = DateTime.UtcNow;
+
+        _context.Articles.Update(article);
+        await _context.SaveChangesAsync();
+        
+        var articleDto = _mapper.Map<ArticleDto>(article);
+
+        return articleDto;
+    }
+    
+    public async Task DeleteByIdAsync(int id, JwtInfoDto userToken)
+    {
+        var article = await GetArticleByIdAsync(id);
+
+        ThrowIfUserHasNotEditAccess(article.UserId, userToken);
+        
+        _context.Articles.Remove(article);
+        await _context.SaveChangesAsync();
+    }
+
+    
+    
+    private async Task<Article> GetArticleByIdAsync(int id)
+    {
+        var article = await _context.Articles.FirstOrDefaultAsync(article => article.Id == id) 
+                      ?? throw new NotFoundException($"Article with Id: {id} is not found");
+
+        return article;
+    }
+
+    private async Task ThrowIfTitleExistAsync(ArticleDtoInput articleInput)
+    {
+        if (await _context.Articles.AnyAsync(article => article.Title == articleInput.Title))
+        {
+            throw new BadRequestException($"Article with Title: {articleInput.Title} exists");
+        }
+    }
+    
+    private async Task ThrowIfTitleExistAsync(ArticleDtoInput articleInput, int id)
+    {
+        var existArticle = await GetArticleByIdAsync(id);
+
+        if (await _context.Articles.AnyAsync(article =>
+                article.Title == articleInput.Title && article.Id != existArticle.Id))
+        {
```

Comment:
do not split if and foreach cases (if it is too long, save it on the local variable and use this one in the if case)